adafruit_nunchuk.py

```diff
@@ -61,32 +67,66 @@ def __init__(self, i2c, address=0x52, i2c_read_delay=0.002):
             i2c_dev.write(b"\xFB\x00")
 
     @property
-    def joystick(self):
-        """Return tuple of current joystick position."""
+    def values(self):
+        """The current state of all values."""
         self._read_data()
-        return self.buffer[0], self.buffer[1]
+        return self._Values(
+            self._joystick(do_read=False),
+            self._buttons(do_read=False),
+            self._acceleration(do_read=False),
+        )
 
     @property
-    def button_C(self):  # pylint: disable=invalid-name
-        """Return current pressed state of button C."""
-        return not bool(self._read_data()[5] & 0x02)
+    def joystick(self):
+        """The current joystick position."""
+        return self._joystick()
 
     @property
-    def button_Z(self):  # pylint: disable=invalid-name
-        """Return current pressed state of button Z."""
-        return not bool(self._read_data()[5] & 0x01)
+    def buttons(self):  # pylint: disable=invalid-name
+        """The current pressed state of button Z."""
+        return self._buttons()
 
     @property
     def acceleration(self):
-        """Return 3 tuple of accelerometer reading."""
-        self._read_data()
-        x = (self.buffer[5] & 0xC0) >> 6
-        x |= self.buffer[2] << 2
-        y = (self.buffer[5] & 0x30) >> 4
-        y |= self.buffer[3] << 2
-        z = (self.buffer[5] & 0x0C) >> 2
-        z |= self.buffer[4] << 2
-        return x, y, z
+        """The current accelerometer reading."""
+        return self._acceleration()
+
+    @property
+    def button_C(self):  # pylint: disable=invalid-name
+        """
+        The current pressed state of button C.
+        """
+        print("`button_C` is deprecated. Please use the `buttons` property instead")
+        return self._buttons().C
+
+    @property
+    def button_Z(self):  # pylint: disable=invalid-name
+        """
+        The current pressed state of button Z.
+        """
+        print("`button_Z` is deprecated. Please use the `buttons` property instead")
+        return self._buttons().Z
```

Comment:
Let's go ahead and break the API and remove `button_C` and `button_Z`. The new way will be to use the single `buttons` property for either button.